src/boot/bootctl-uki.c

```diff
@@ -143,21 +144,49 @@ static int read_pe_section(FILE *uki, const struct PeSectionHeader *section,
         return 0;
 }
 
+static int inspect_osrel(char *osrel, size_t osrel_size) {
+        _cleanup_fclose_ FILE *s = NULL;
+        _cleanup_free_ char *pname = NULL, *name = NULL;
+        int r;
+
+        assert(osrel);
+        s = fmemopen(osrel, osrel_size, "r");
+        if (!s)
+                return log_error_errno(errno, "Failed to open embedded os-release file, ignoring: %m");
+
+        r = parse_env_file(s, NULL,
+                           "PRETTY_NAME", &pname,
+                           "NAME",        &name);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse embedded os-release file, ignoring: %m");
+
+        if (pname || name)
+                printf("         OS: %s\n", pname ?: name);
```

Comment:
#26158 unifies this across the codebase. Once that is landed this should be changed to just do:

```
printf("      OS: %s\n", os_release_pretty_name(pname, name));
```

even the conditionalization could go, as it will then print "Linux".